### PR TITLE
fix: fix type casting issue

### LINKED ISSUE #1594: `CAST` data type is needed when using `COPY FROM`
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Frontend

### What happened?

Cannot import csv file via `COPY FROM`

### What operating system did you use?

Mac

### Relevant log output and stack trace

_No response_

### How can we reproduce the bug?

The data example:

```
hostname,environment,usage_user,usage_system,usage_idle,usage_nice,usage_iowait,usage_irq,usage_softirq,usage_steal,usage_guest,usage_guest_nice,ts
host_0,test,32,58,36,72,61,21,53,12,59,72,2023-04-01T00:00:00+00:00
host_1,staging,12,32,50,84,19,73,38,37,72,2,2023-04-01T00:00:00+00:00
host_2,test,98,5,40,95,64,39,21,63,53,94,2023-04-01T00:00:00+00:00
host_3,test,98,95,7,48,99,67,14,86,36,23,2023-04-01T00:00:00+00:00
host_4,test,32,44,11,53,64,9,17,39,20,7,2023-04-01T00:00:00+00:00
```

Then create table:
```
create table tsbs_cpu(hostname STRING, environment STRING, usage_user DOUBLE, usage_system DOUBLE, usage_idle DOUBLE, usage_nice DOUBLE, usage_iowait DOUBLE, usage_irq DOUBLE, usage_softirq DOUBLE, usage_steal DOUBLE, usage_guest DOUBLE, usage_guest_nice DOUBLE, ts TIMESTAMP TIME INDEX, PRIMARY KEY(hostname));
```

Import data:

```
copy tsbs_cpu from '/tmp/greptimedb/tom.csv' WITH(FORMAT='csv');
```

get the error:

> ERROR 1815 (HY000): Failed to execute query: copy tsbs_cpu from '/tmp/greptimedb/tom.csv' WITH(FORMAT='csv'), source: Schema datatypes not match at index 2, expected table schema: Field { name: "hostname", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "environment", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "usage_user", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name

I guess the reason is the schema doesn't match, since the type of `usage_user` is double, but my data value is `32`.

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Fix incorrect type-casting check

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
fix #1594

### CODE REVIEW

File: src/frontend/src/statement/copy_table_from.rs [hunk 1]
```diff
@@ -519,17 +464,17 @@ mod tests {
                 ),
                 true,
             ),
-            false,
+            true,
```
Comment (WenyXu) [inline src/frontend/src/statement/copy_table_from.rs:467]: Allows type cast from `Second` to  `Millisecond `

File: src/common/datasource/tests/csv/type_cast.csv [hunk 2]
```diff
@@ -0,0 +1,6 @@
+hostname,environment,usage_user,usage_system,usage_idle,usage_nice,usage_iowait,usage_irq,usage_softirq,usage_steal,usage_guest,usage_guest_nice,ts
+host_0,test,32,58,36,72,61,21,53,12,59,72,2023-04-01T00:00:00+00:00
+host_1,staging,12,32,50,84,19,73,38,37,72,2,2023-04-01T00:00:00+00:00
+host_2,test,98,5,40,95,64,39,21,63,53,94,2023-04-01T00:00:00+00:00
+host_3,test,98,95,7,48,99,67,14,86,36,23,2023-04-01T00:00:00+00:00
+host_4,test,32,44,11,53,64,9,17,39,20,7,2023-04-01T00:00:00+00:00
```
Comment (v0y4g3r) [inline src/common/datasource/tests/csv/type_cast.csv:6]: Missing new line at the end of file.

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**